lua/terrortown/autorun/shared/sh_roundinfo_convars.lua

```diff
@@ -1,63 +1,9 @@
-	ULib.replicatedWritableCvar("ttt_roundinfo_popup_killer_time", "rep_ttt_roundinfo_popup_killer_time", GetConVar("ttt_roundinfo_popup_killer_time"):GetInt(), true, false, name)
-	ULib.replicatedWritableCvar("ttt_roundinfo_post_announce_distribution", "rep_ttt_roundinfo_post_announce_distribution", GetConVar("ttt_roundinfo_post_announce_distribution"):GetInt(), true, false, name)
-end)
-
 if SERVER then
 	AddCSLuaFile()
 end
 
-if CLIENT then
-	hook.Add("TTTUlxModifyAddonSettings", "TTTRolesetupModifySettings", function(name)
-		local tttrspnl = xlib.makelistlayout{w = 415, h = 318, parent = xgui.null}
-
-		-- Chat Messages
-		local tttrsclp = vgui.Create("DCollapsibleCategory", tttrspnl)
-		tttrsclp:SetSize(390, 80)
-		tttrsclp:SetExpanded(1)
-		tttrsclp:SetLabel("Chat Massages")
-
-		local tttrslst = vgui.Create("DPanelList", tttrsclp)
-		tttrslst:SetPos(5, 25)
-		tttrslst:SetSize(390, 80)
-		tttrslst:SetSpacing(5)
-
-		local tttrsdh1 = xlib.makecheckbox{label = "ttt_roundinfo_pre_announce_distribution (Def. 1)", repconvar = "rep_ttt_roundinfo_pre_announce_distribution", parent = tttrslst}
-		tttrslst:AddItem(tttrsdh1)
-
-		local tttrsdh2 = xlib.makecheckbox{label = "ttt_roundinfo_announce_killer (Def. 1)", repconvar = "rep_ttt_roundinfo_announce_killer", parent = tttrslst}
-		tttrslst:AddItem(tttrsdh2)
-
-		local tttrsdh3 = xlib.makecheckbox{label = "ttt_roundinfo_post_announce_distribution (Def. 1)", repconvar = "rep_ttt_roundinfo_post_announce_distribution", parent = tttrslst}
-		tttrslst:AddItem(tttrsdh3)
-
-		-- Popup
-		local tttrsclp2 = vgui.Create("DCollapsibleCategory", tttrspnl)
-		tttrsclp2:SetSize(390, 60)
-		tttrsclp2:SetExpanded(1)
-		tttrsclp2:SetLabel("Popup")
-
-		local tttrslst2 = vgui.Create("DPanelList", tttrsclp2)
-		tttrslst2:SetPos(5, 25)
-		tttrslst2:SetSize(390, 60)
-		tttrslst2:SetSpacing(5)
-
-		local tttrsdh4 = xlib.makecheckbox{label = "ttt_roundinfo_popup_killer (Def. 1)", repconvar = "rep_ttt_roundinfo_popup_killer", parent = tttrslst}
-		tttrslst2:AddItem(tttrsdh4)
-
-		local tttrsdh5 = xlib.makeslider{label = "ttt_roundinfo_popup_killer_time (Def. 10)", repconvar = "rep_ttt_roundinfo_popup_killer_time", min = 1, max = 100, decimal = 0, parent = tttrslst}
-		tttrslst2:AddItem(tttrsdh5)
-
-		xgui.hookEvent("onProcessModules", nil, tttrspnl.processModules)
-		xgui.addSubModule("Round Info", tttrspnl, nil, name)
-	end)
-end
\ No newline at end of file
+CreateConVar("ttt_roundinfo_pre_announce_distribution", 1, {FCVAR_NOTIFY, FCVAR_ARCHIVE, FCVAR_REPLICATED})
+CreateConVar("ttt_roundinfo_announce_killer", 1, {FCVAR_NOTIFY, FCVAR_ARCHIVE, FCVAR_REPLICATED})
+CreateConVar("ttt_roundinfo_popup_killer", 1, {FCVAR_NOTIFY, FCVAR_ARCHIVE, FCVAR_REPLICATED})
+CreateConVar("ttt_roundinfo_popup_killer_time", 10, {FCVAR_NOTIFY, FCVAR_ARCHIVE, FCVAR_REPLICATED})
+CreateConVar("ttt_roundinfo_post_announce_distribution", 1, {FCVAR_NOTIFY, FCVAR_ARCHIVE, FCVAR_REPLICATED})
```

Comment:
As this will be server only and we dont need the replicatedCvar anymore you could move that to the correct file.
Not sure if there is a sv_roundinfo or we rather make an `if CLIENT return`-statement.

Reply:
client needs the convar defined for replication to work, and for the value to be read by the F1 menu, right?

Reply:
Mhh, Im actually not sure anymore. I thought client doesnt need it.

Reply:
I believe that my F1 menu serverConvar handles that. But I'm gonna check that later again.

Reply:
the menu does not need replicated convars

Reply:
I'd remove that file and put these convar creations in one of the server files and call it a day

Reply:
> the menu does not need replicated convars

[i'm getting mixed signals here](https://github.com/TTT-2/ttt2-wep_minifier/pull/4#issuecomment-1900246435)

Reply:
Client UI doesnt mean F1 menu. The F1 menu uses an internal syncing system before replicated convars worked.
I believe what Tim meant was ingame UI showing states of the used equipment.
And depending on how I implemented it, you can probably also use our TTT2 serverConvar syncing for that. But instead it still checks ConVars 🤔